src/libServer/Server.cpp

```diff
@@ -60,7 +68,52 @@ string Server::getProtocolVersion()
 
 string Server::createTransaction(const Json::Value& _json)
 {
-	return "Hello";
+	LOG_MARKER();
+
+	if(!JSONConversion::checkJsonTx(_json))
+	{
+		return "Invalid Tx Json";
+	}
+
+	Transaction tx = JSONConversion::convertJsontoTx(_json);
+ 
+	//LOG_MESSAGE("Nonce: "<<tx.GetNonce().str()<<" toAddr: "<<tx.GetToAddr().hex()<<" senderPubKey: "<<static_cast<string>(tx.GetSenderPubKey());<<" amount: "<<tx.GetAmount().str());
+
+    unsigned int num_shards = m_mediator.m_lookup->GetShardPeers().size();
+
+    
+    const PubKey & senderPubKey = tx.GetSenderPubKey();
+    const Address fromAddr = Account::GetAddressFromPublicKey(senderPubKey);
+    unsigned int curr_offset = 0;
+    
+    if(num_shards>0)
+    {
+    	unsigned int shard = Transaction::GetShardIndex(fromAddr, num_shards);
+    	map <PubKey, Peer> shardMembers = m_mediator.m_lookup->GetShardPeers().at(shard);
+    	LOG_MESSAGE("The Tx Belongs to "<<shard<<" Shard");
+
+    	vector<unsigned char> tx_message = {MessageType::NODE, NodeInstructionType::CREATETRANSACTIONFROMLOOKUP};
+    	curr_offset += MessageOffset::BODY;
+
+    	tx.Serialize(tx_message, curr_offset);
+
+    	LOG_MESSAGE("Tx Serialized");
+
+
+    	P2PComm::GetInstance().SendMessage(shardMembers.begin()->second, tx_message);
```

Comment:
Instead of sending to the all members in the shard, randomly pick 20 and send to only these 20 nodes in the shard.

Reply:
Right now, it is only sending to one

Reply:
Changes : send to first 20 nodes, if less than 20 send it to all. From discussion with @bb111189